x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java

```diff
@@ -35,4 +43,53 @@ public String toString() {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.value(shapeBuilder.toWKT());
     }
+
+    public int dimension() {
+        return shapeBuilder.inherentDimensions();
+    }
+
+    public String geometryType() {
+        return geometryType(shapeBuilder.type());
+    }
+
+    static String geometryType(GeoShapeType type) {
+        switch (type) {
+            case POINT: return "Point";
+            case CIRCLE: return "Circle";
+            case POLYGON: return "Polygon";
+            case ENVELOPE: return "Envelope";
+            case LINESTRING: return "LineString";
+            case MULTIPOINT: return "MultiPoint";
+            case MULTIPOLYGON: return "MultiPolygon";
+            case MULTILINESTRING: return "MultiLineString";
+            case GEOMETRYCOLLECTION: return "GeometryCollection";
+            default:
+                throw new SqlIllegalArgumentException("Unsupported geometry type [" + type + "]");
+
+        }
+    }
+
+    public double x() {
+        return shapeBuilder.firstX();
+    }
+
+    public double y() {
+        return shapeBuilder.firstY();
+    }
+
+    public double minX() {
+        return shapeBuilder.buildS4J().getBoundingBox().getMinX();
```

Comment:
@nknize that's probably not the best way to handle this. Do you have any suggestions on what can be done here?